File: Scripts/laptop/function-composition/c-sharp/function-composition.cs
using System;

class Program
{

    private static volatile int r = 1;
    
    public static int executeTask(int i) {

        Func<float, float> outfunc = Composer<float, float, float>.Compose(functA, functB);
	outfunc((float)i);
 	return 1 + i;
    } 

    static void Main(string[] args)
    {
        for (int i = 0; i < 1000000000; ++i)
	{
		r = executeTask(i + r);
    	}
	}
  	static float functA(float i) { return (float)Math.Sin(i); }
	static float functB(float i) { return (float)Math.Asin(i); }


    class Composer<A, B, C>
    {
        public static Func<C, A> Compose(Func<B, A> a, Func<C, B> b)
        {
            return delegate(C i) { return a(b(i)); };
        }
    }
}

File: Scripts/volatile_and_force_compiler_task/function-composition/python/function-composition.py
from numpy import *
from math import sin, asin

def executeTask(i):
    compose = lambda f, g: lambda x: f( g(x) )
    sin_asin = compose(sin, asin)
    i = i +1
    return sin_asin(1/i)

r = 0.0
for i in xrange(1000000000):
	r = executeTask(i)
